runtime/health.go

```diff
@@ -46,7 +46,8 @@ func (r *Runtime) Health(ctx context.Context) (*Health, error) {
 
 	ih := make(map[string]*InstanceHealth, len(instances))
 	for _, inst := range instances {
-		ih[inst.ID], err = r.InstanceHealth(ctx, inst.ID)
+		// if there is a single instance hosted on this runtime then instead of returning error msgs throw error if OLAP/repo/controller are in error state
+		ih[inst.ID], err = r.InstanceHealth(ctx, inst.ID, len(instances) == 1)
```

Comment:
nit: the `retErr` option maybe saves a few lines of code, but I think it might be nicer to keep the signature and behavior simple, and explicitly handle the desired error cases here, e.g.:
```suggestion
		ih[inst.ID], err = r.InstanceHealth(ctx, inst.ID)
		if err != nil ... {}
		
		// If there is single instance ....
		if len(instances) == 1 {
			h := ih[inst.ID]
			if h.Controller != "" {
				return ...
			}
			if h.OLAP != "" {
				return ...
			}
			// etc.
		}
```

Reply:
This is also fine just that need to duplicate this handling at two places. Here as well as API handler.